### PR TITLE
feat(vim.version): add `vim.version.le` and `vim.version.ge`

### PR DESCRIPTION
**Pull request by wookayin**
- Problem: One cannot easily write something like, for example:
    `version_current >= {0, 10, 0}`; writing like
    `not vim.version.lt(version_current, {0, 10, 0})` is verbose.

- Solution: add {`le`,`ge`} in addition to {`lt`,`gt`}.

- Also improve typing on the operator methods: allow `string` as well.

- Update the example in `vim.version.range()` docs: `ge` in place of
  `gt` better matches the semantics of `range:has`.

### CODE REVIEW

File: test/functional/lua/version_spec.lua [hunk 1]
```diff
@@ -288,21 +288,55 @@ describe('version', function()
...
   it('eq()', function()
     eq(true, vim.version.eq('2', '2'))
     eq(true, vim.version.eq({ 3, 1, 0 }, '3.1.0'))
     eq(true, vim.version.eq({ major = 3, minor = 3, patch = 0 }, { 3, 3, 0 }))
+    eq(false, vim.version.eq('2', '3'))
+
+    -- semver: v3 == v3.0 == v3.0.0
+    eq(true, vim.version.eq('3', { 3, 0, 0 }))
+    eq(true, vim.version.eq({ 3, 0 }, { 3 }))
+    eq(true, vim.version.eq({ 3, 0, 0 }, { 3 }))
```
[the remark this answers is not in the document]
Comment (dundargoc) [reply test/functional/lua/version_spec.lua:340]: ACKSHUALLY, a prerelease is supposed to be separated by a hyphen according to https://semver.org/#spec-item-9 🤓. The second variant should segfault or something

[the remark this answers is not in the document]
Comment (dundargoc) [reply test/functional/lua/version_spec.lua:340]: But yes, otherwise I think that is incorrect if I read the semver page correctly.

**Review by dundargoc — APPROVED** [review]
Looks good an make sense IMO.

File: test/functional/lua/version_spec.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (wookayin) [reply test/functional/lua/version_spec.lua:340]: Looking at the implementation, the fourth item and thereafter are simply ignored. So checking the length (1~3) would be needed.

[the remark this answers is not in the document]
Comment (dundargoc) [reply test/functional/lua/version_spec.lua:340]: I was being serious by the way, `3.0.0.999` is not semver and the above is not a valid comparison.

**Comment by wookayin** [comment]
@dundargoc @lewis6991 LGTY then let's merge?

**Pull request merged**